File: Samples/VB/10-CheckBox/Form1.vb
Imports RibbonLib
Imports RibbonLib.Controls
Imports RibbonLib.Controls.Events
Imports RibbonLib.Interop

Namespace _10_CheckBox
	Public Enum RibbonMarkupCommands As UInteger
		 cmdButton = 1001
		 cmdToggleButton = 1002
		 cmdCheckBox = 1003
		 cmdDropDown = 1004
		 cmdTabMain = 1011
		 cmdGroupButtons = 1013
		 cmdGroupCheckBox = 1014
	End Enum

	Partial Public Class Form1
		Inherits Form
		Private _button As RibbonButton
		Private _toggleButton As RibbonToggleButton
		Private _checkBox As RibbonCheckBox

		Public Sub New()
			InitializeComponent()

			_button = New RibbonButton(_ribbon, CUInt(RibbonMarkupCommands.cmdButton))
			_toggleButton = New RibbonToggleButton(_ribbon, CUInt(RibbonMarkupCommands.cmdToggleButton))
			_checkBox = New RibbonLib.Controls.RibbonCheckBox(_ribbon, CUInt(RibbonMarkupCommands.cmdCheckBox))

			AddHandler _button.ExecuteEvent, AddressOf _button_ExecuteEvent
		End Sub

		Private Sub _button_ExecuteEvent(ByVal sender As Object, ByVal e As ExecuteEventArgs)
			MessageBox.Show("checkbox check status is: " & _checkBox.BooleanValue.ToString())
		End Sub
	End Class
End Namespace

File: Samples/CS/10-CheckBox/Form1.cs
using System;
using System.Windows.Forms;
using RibbonLib;
using RibbonLib.Controls;
using RibbonLib.Controls.Events;
using RibbonLib.Interop;

namespace _10_CheckBox
{
    public enum RibbonMarkupCommands : uint 
    {
         cmdButton = 1001,
         cmdToggleButton = 1002,
         cmdCheckBox = 1003,
         cmdDropDown = 1004,
         cmdTabMain = 1011,
         cmdGroupButtons = 1013,
         cmdGroupCheckBox = 1014,
    }

    public partial class Form1 : Form
    {
        private RibbonButton _button;
        private RibbonToggleButton _toggleButton;
        private RibbonCheckBox _checkBox;

        public Form1()
        {
            InitializeComponent();

            _button = new RibbonButton(_ribbon, (uint)RibbonMarkupCommands.cmdButton);
            _toggleButton = new RibbonToggleButton(_ribbon, (uint)RibbonMarkupCommands.cmdToggleButton);
            _checkBox = new RibbonLib.Controls.RibbonCheckBox(_ribbon, (uint)RibbonMarkupCommands.cmdCheckBox);

            _button.ExecuteEvent += new EventHandler<ExecuteEventArgs>(_button_ExecuteEvent);
        }

        void _button_ExecuteEvent(object sender, ExecuteEventArgs e)
        {
            MessageBox.Show("checkbox check status is: " +  _checkBox.BooleanValue.ToString());
        }
    }
}
